lib/pbench/server/database/models/users.py

```diff
@@ -81,6 +74,13 @@ def update(self, **kwargs):
                     # Insert the auth token
                     self.auth_tokens.append(value)
                     Database.db_session.add(value)
+                elif key == "password":
+                    setattr(
+                        self, key, generate_password_hash(value, self.bcrypt_log_rounds)
+                    )
+                # Prevent update on "registered_on" and "bcrypt_log_rounds" fields
+                elif key in ["registered_on", "bcrypt_log_rounds"]:
+                    continue
```

Comment:
OK; so if you modify two properties and PUT the entire new object (which is the way a REST update would normally work), we silently ignore non-updateable properties. That's good here, but I think we also want to diagnose an attempt (probably at PUT rather than here) to change properties you're not allowed to change... which means "OK, `registered_on` is in the payload, and that's OK if the value hasn't changed, but reported as an error in the value is different.".